js/app.js

```diff
@@ -174,9 +174,31 @@ function injectPage(url, opt_addToHistory) {
   xhr.send();
 }
 
+// Old API reference URLs have the page name in the hash. After
+// server-side redirects, they end up as "docs/elements/#page-name"
+// The page name may be followed by a deep link, like ".attributes.data".
+// Rewrite here to "docs/elements/page-name.html", leaving any hash
+// in place to preserve the deep link.
+function redirectOldAPIDocs() {
+  var oldAPILanding = 'docs/elements/'
+  var path = window.location.pathname;
+  var hash = window.location.hash;
+  var position = path.length - oldAPILanding.length;
+  var lastIndex =  path.indexOf(oldAPILanding, position);
+  if (lastIndex !== -1 && lastIndex == position) {
+    if (hash) {
+      location.href = location.href.replace(/(\/docs\/elements\/)#([^.]*)(.*)$/, '$1$2.html#$2$3')
```

Comment:
Yeah, it will. URLs like:

http://localhost:3000/0.5/docs/elements/#core-list

Become: 

http://localhost:3000/0.5/docs/elements/core-list.html#core-list

The extra hash doesn't do anything, so I didn't add a special case for it. But I could add:

if (hash.indexOf('.')) { ... } else { a different regex } ... ?